src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1406,11 +1407,15 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
                     f'`.{fn}(ckpt_path="best")` is set but `ModelCheckpoint` is not configured to save the best model.'
                 )
             # load best weights
-            ckpt_path = self.checkpoint_callback.best_model_path
+            ckpt_path = (
+                self.checkpoint_callback.best_model_path
+                if hasattr(self.checkpoint_callback, "best_model_path")
+                else None
+            )
 
         if ckpt_path == "last":
-            candidates = [ft.ckpt_path for ft in ft_checkpoints] + [
-                cb.last_model_path for cb in self.checkpoint_callbacks
+            candidates = [ft.ckpt_path if hasattr(ft, "ckpt_path") else None for ft in ft_checkpoints] + [
+                cb.last_model_path if hasattr(cb, "last_model_path") else None for cb in self.checkpoint_callbacks
```

Comment:
```suggestion
            candidates = [getattr(ft, "ckpt_path", None) for ft in ft_checkpoints] + [
                getattr(cb, "last_model_path", None) for cb in self.checkpoint_callbacks
```

Reply:
Oh my, I feel dumb for using this antipattern. Thanks @awaelchli for pointing it out :)